overlord/state/warning.go

```diff
@@ -0,0 +1,246 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package state
+
+import (
+	"encoding/json"
+	"sort"
+	"time"
+)
+
+var (
+	DefaultRepeatAfter = time.Hour * 24
+	DefaultDeleteAfter = time.Hour * 24 * 28
+)
+
+type jsonWarning struct {
+	Message     string     `json:"message"`
+	FirstSeen   time.Time  `json:"first-seen"`
+	LastSeen    time.Time  `json:"last-seen"`
+	LastShown   *time.Time `json:"last-shown,omitempty"`
+	DeleteAfter string     `json:"delete-after,omitempty"`
+	RepeatAfter string     `json:"repeat-after,omitempty"`
+}
+
+type Warning struct {
+	// the warning text itself. Only one of these in the system at a time.
```

Comment:
Can you explain what you meant by `only one of these in the system at a time`. Does it mean that only one warning saying `Yada yada the Spanish inquisition` is ever stored in the state?

Reply:
yes